src/Extension/FluentVersionedExtension.php

```diff
@@ -595,4 +634,331 @@ public function updateLocalisationTabConfig(GridFieldConfig $config)
             PublishAction::create(),
         ]);
     }
+
+    /**
+     * Localise archived state
+     * Extension point in @see Versioned::isArchived()
+     *
+     * @param bool $isArchived
+     */
+    public function updateIsArchived(bool &$isArchived): void
+    {
+        $locale = FluentState::singleton()->getLocale();
+
+        if (!$locale) {
+            return;
+        }
+
+        $isArchived = $this->owner->isArchivedInLocale();
+    }
+
+    /**
+     * @param string|null $locale
+     * @return bool
+     */
+    public function isArchivedInLocale(?string $locale = null): bool
+    {
+        $locale = $locale ?: FluentState::singleton()->getLocale();
+
+        if (!$locale) {
+            return false;
```

Comment:
I'm wondering if perhaps this should return `null`, rather than `false`?

`false` indicates that something is specifically **not** archived in a locale, but if there is no locale, then I think neither `true` or `false` is technically accurate.

This should also mean that `updateIsArchived()` does not update the flag at all - whatever was determined previously is left as is.